pulpcore/app/tasks/export.py

```diff
@@ -74,18 +77,38 @@ def pulp_export(pulp_exporter):
     Args:
         pulp_exporter (models.PulpExporter): PulpExporter instance
     """
-    # import pydevd_pycharm
-    # pydevd_pycharm.settrace('192.168.1.109', port=3014, stdoutToServer=True, stderrToServer=True)
-
     repositories = pulp_exporter.repositories.all()
     export = PulpExport.objects.create(exporter=pulp_exporter, task=Task.current(), params=None)
     # TODO: 'soon' path will be a tarfile and we will be streaming directly into the tarfile
     export.destination_dir = os.path.join(pulp_exporter.path, export_dir(export))
     CreatedResource.objects.create(content_object=export)
 
+    # 1) Spit out all Artifacts
+    # 2) Spit out artifactresource JSON limited by the ablove
+    # 3) Spit out one repositoryresource JSON w/all repositories
+    # 4) Spit out all *resource JSONs per-repo-version
+
+    artifacts = []
+    repo_versions = []
+    # Gather up the versions and artifacts
     for repo in repositories:
         version = repo.latest_version()
-        export.export_repository_version(version)
+        # Check version-content to make sure we're not being asked to export an on_demand repo
+        content_artifacts = ContentArtifact.objects.filter(content__in=version.content)
+        if content_artifacts.filter(artifact=None).exists():
+            RuntimeError(_("Remote artifacts cannot be exported."))
+
+        repo_versions.append(version)
+        for ca in content_artifacts:
```

Comment:
This is a horribly-inefficient way to get all the Artifacts for a repo-version - what's the Correct Way?